test/cql-pytest/test_permissions.py

```diff
@@ -316,12 +316,12 @@ def test_grant_perms_on_nonexistent_udf(cql):
         revoke(cql, 'EXECUTE', 'ALL FUNCTIONS', username)
         with pytest.raises(InvalidRequest):
             grant(cql, 'EXECUTE', f'ALL FUNCTIONS IN KEYSPACE {keyspace}', username)
-        with pytest.raises(InvalidRequest):
+        with pytest.raises((InvalidRequest, ConfigurationException)):
```

Comment:
The discussion in test/cql-pytest/test_keyspace.py test_drop_keyspace_nonexistent seems interesting - when dropping a nonexistent keyspace Cassandra used to return a ConfigurationException and in 4.0 they modified it to a InvalidRequest exception, while we keep returning the ConfigurationException.
In this UDF test, however, we return the InvalidRequest, and Cassandra still returns ConfigurationException. I think Cassandra must've forgotten to update the exception type here, because when granting a permission on a table, they also return an InvalidRequest.
I can add some of the discussion above to a comment, but I'm not sure whether this is something that needs a fix in terms of the implementation